File: backend/src/main/java/live/dilmith/crypticbrain/backend/entity/Room.java
package live.dilmith.crypticbrain.backend.entity;

import jakarta.persistence.*;
import live.dilmith.crypticbrain.backend.enums.RoomStatus;
import live.dilmith.crypticbrain.backend.enums.RoomType;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "rooms")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true)
    private String roomNumber;

    @Enumerated(EnumType.STRING)
    private RoomType roomType;

    private Integer floor;

    private Integer capacity;

    private BigDecimal pricePerNight;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private RoomStatus status = RoomStatus.AVAILABLE;

    @Column(nullable = true)
    private String description;

    @Column(nullable = true)
    private String amenities;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/room.ts
export type RoomType = 'SINGLE' | 'DOUBLE' | 'SUITE' | 'DELUXE';
export type RoomStatus = 'AVAILABLE' | 'OCCUPIED' | 'UNDER_MAINTENANCE' | 'CLEANING';

export interface Room {
  id: number;
  roomNumber: string;
  roomType: RoomType;
  floor: number;
  capacity: number;
  pricePerNight: number;
  status: RoomStatus;
  description?: string;
  amenities?: string;
  createdAt: string;
  updatedAt: string;
}

export interface RoomCreateRequest {
  roomNumber: string;
  roomType: RoomType;
  floor: number;
  capacity: number;
  pricePerNight: number;
  status?: RoomStatus;
  description?: string;
  amenities?: string;
}

export interface RoomUpdateRequest {
  roomNumber?: string;
  roomType?: RoomType;
  floor?: number;
  capacity?: number;
  pricePerNight?: number;
  status?: RoomStatus;
  description?: string;
  amenities?: string;
}
